packages/common/src/utils.js

```diff
@@ -256,6 +256,56 @@ function cleanChildrenRoutes(routes, isChild = false) {
   return routes
 }
 
+const DYNAMIC_ROUTE_REGEX = /^\/(:|\*)/
+
+const sortRoutes = function sortRoutes(routes) {
+  routes.sort((a, b) => {
+    if (!a.path.length) {
+      return -1
+    }
+    if (!b.path.length) {
+      return 1
+    }
+    // Order: /static, /index, /:dynamic
+    // Match exact route before index: /login before /index/_slug
+    if (a.path === '/') {
+      return DYNAMIC_ROUTE_REGEX.test(b.path) ? -1 : 1
+    }
+    if (b.path === '/') {
+      return DYNAMIC_ROUTE_REGEX.test(a.path) ? 1 : -1
+    }
+
+    let i
+    let res = 0
+    let y = 0
+    let z = 0
+    const _a = a.path.split('/')
+    const _b = b.path.split('/')
+    for (i = 0; i < _a.length; i++) {
+      if (res !== 0) {
+        break
+      }
+      y = _a[i] === '*' ? 2 : _a[i].includes(':') ? 1 : 0
+      z = _b[i] === '*' ? 2 : _b[i].includes(':') ? 1 : 0
+      res = y - z
+      // If a.length >= b.length
+      if (i === _b.length - 1 && res === 0) {
+        // sort alphabetically unless * found
+        res = _a[i] === '*' ? -1 : a.path.localeCompare(b.path)
```

Comment:
Should `a.path.localeCompare(b.path)` be `b.path.localeCompare(a.path)` ?

Since `'snake_case_route'.localeCompare('another_route-id')` is `1`, but we need `another_route-id`is in front of `snake_case_route` as alphabetical order.

Reply:
The tests succeeds in v11 actually but fails in v10. It is because the last line in sortRoutes also has a fallback to sort by default with -1 regardless of value, I missed that because I tested with `dynamic-routes` and it seems that test doesnt test that code path. Currently trying to fix the dynamic-routes test to include that ;)

Reply:
`(_a[i - 1] === '*' && _b[i] ? 1 : -1)` could be `(_a[i - 1] === '*' && _b[i] ? 1 : _a[i - 1].localeCompare(_b[i - 1]))` ?

Reply:
If we use that then `/test/projects` is sorted before `/test`, if I use `a.path.localeCompare(b.path)` then `/test/songs/toto` is sorted before `/test/users`, so I think we should only compare alphabetically if _a.length === _b.length